[Refacto] Mettre le layout de Pix Site en grid

Anne-Gaelle-S opened the pull request:
## :unicorn: Problème
Les maquettes de pix-site sont faites à partir de grilles adaptatives.
Techniquement nous utilisons des marges internes / le module flex pour positionner notre layout général, cependant cela complexifie l'implémentation des maquettes.

## :robot: Solution
Utiliser le module [CSS grid layout](https://developer.mozilla.org/fr/docs/Web/CSS/CSS_Grid_Layout).
Cela améliora ces deux points : 
- la communication avec nos designers (puisqu'ils travaillent avec un support de grid)
- l'implémentation de la mise en page du site

## :rainbow: Remarques
Styles réutilisable pour le layout général : 
**1- Contenu principal à GAUCHE & contenu accompagnant à DROITE**
```
<div class="row-block">
   <div class="row-block__left-content"></div>
   <div class="row-block__right-content"></div>
</div>
```
Exemple : 
<img alt="image" src="https://user-images.githubusercontent.com/38167520/90150348-d90b3280-dd85-11ea-90db-69c72b7f0259.png">

**2- Contenu principal à DROITE & contenu accompagnant à GAUCHE**
```
<div class="row-block row-block--reverse">
   <div class="row-block__left-content"></div>
   <div class="row-block__right-content"></div>
</div>
```
Exemple : 

Large-screen |  Mobile
:-------------------------:|:-------------------------:
<img  alt="image" src="https://user-images.githubusercontent.com/38167520/90150409-ec1e0280-dd85-11ea-9e1f-1be3b213127e.png"> | <img alt="image" src="https://user-images.githubusercontent.com/38167520/90150540-1079df00-dd86-11ea-9314-95697697428b.png">

Pour les maquettes voir : https://app.abstract.com/projects/c43b8749-ba3a-474e-a6c5-dbf76bade2ee/branches/master/collections/016f15c0-b9b4-4886-99be-dcc9d4f5fa60

## 🐢 Spéciale dédicace à notre ami IE
Pour que le module `grid layout` de css fonctionne sous IE il est pratique d'utiliser [autoprefixer](https://github.com/postcss/autoprefixer#does-autoprefixer-polyfill-grid-layout-for-ie).
Cet outil, que dis-je, cette **PEPITE** permet de rajouter AUTOMATIQUEMENT 🧙‍♂️ les préfixes nécessaires pour que les grid fonctionne sous tous les navigateurs (dans la mesure du possible).

Résultat des grids sous IE : 
<img width="1669" alt="image" src="https://user-images.githubusercontent.com/38167520/90155885-238fad80-dd8c-11ea-8243-94d8c00809ed.png">


🚧 ATTENTION 
> Autoprefixer only prefixes grid-gap if both grid-template-areas and grid-template-columns have been defined. It also cannot inherit grid-gap through media queries.
[Voir source](https://css-tricks.com/css-grid-in-ie-css-grid-and-the-new-autoprefixer/#table-disclaimers)

Tuto autoprefixer : 
https://css-tricks.com/css-grid-in-ie-css-grid-and-the-new-autoprefixer/

Configuration de l'autoprefixer avec Nuxt : 
https://fr.nuxtjs.org/faq/postcss-plugins/

Pour avoir la liste des browsers que peux préfixer autoprefixer : 
`autoprefixer --info`

## :sparkles: Review App
https://pix-site-review-pr130.osc-fr1.scalingo.io

celineung commented on assets/scss/globals/_buttons.scss:
```diff
@@ -1,121 +1,36 @@
-  &:hover {
-    background-color: $blue-1;
-    color: white !important;
+  &.button-video {
```
du coup, on peut mettre le `hover` des deux types de boutons dans ce fichier?

Anne-Gaelle-S replied:
Yess, c'est ce qui est fait du coup avec le mixin `colorizeButton`: il s'occupe de mettre la couleur de fond, de la police, de la bordure et du hover

celineung commented on assets/scss/globals/_container.scss:
```diff
@@ -29,4 +29,78 @@
+
+  & > * { max-width: 100%; }
+
+  .row-block__main-content  {
+    text-align: center;
+    grid-column: 2 / 6;
+  }
+  .row-block__side-content  { display: none; }
+  .row-block__left-content  { grid-column: 2 / 6; }
+  .row-block__right-content { grid-column: 2 / 6; }
```
cette partie là est très spécifique à `banner` non?

Anne-Gaelle-S replied:
Alors ce qui est spécifique à la banner c'est `.row-block__main-content` et `.row-block__side-content` en effet. 
Mais je me demande s'il y aura pas d'autres pages sur le même principe ? Je vais essayer de demander

Anne-Gaelle-S replied:
Vu avec @QuentinChapelain-ui  en effet la mise en page de la banner est unique (pas répétable ailleurs).
Du coup je change ça :)

Anne-Gaelle-S committed:
Reorganize style in devices sections

Anne-Gaelle-S committed:
Move text/title/button styles in global style

Anne-Gaelle-S committed:
Remove unused styles

Anne-Gaelle-S committed:
Change min-width to show side image on desktop

Anne-Gaelle-S committed:
Install autoprefixer and its configuration for IE

For more information about autoprefixer :
https://github.com/postcss/autoprefixer#does-autoprefixer-polyfill-grid-layout-for-ie

Anne-Gaelle-S committed:
Add grid layout in banner

Anne-Gaelle-S committed:
Add grid global layout with right & left content

Anne-Gaelle-S commented on nuxt.config.js:
```diff
@@ -170,5 +170,19 @@ export default {
+        autoprefixer: {
+          grid: 'autoplace',
+        },
+        browsers: ['last 2 versions', 'ie 11'],
```
Ici on met les versions de browser qu'on souhaite supporter. J'ai mis par défaut les 2 dernières versions de chaque navigateur + ie 11. Peut être qu'on veut en rajouter d'autres ?
Voilà la liste : https://github.com/browserslist/browserslist#full-list

MelanieMEB commented on assets/scss/globals/_container.scss:
```diff
@@ -29,4 +29,72 @@
+  .row-block {
+    grid-template-columns: 32px repeat(8, 1fr) 32px;
+    grid-template-areas:
+      '. a a a a a a a a .'
+      '. b b b b b b b b .';
+
+    &--reverse {
+      grid-template-areas:
+        '. b b b b b b b b .'
+        '. a a a a a a a a .';
+    }
```
Est-ce qu'il ne faudrait pas plus 
```
       '. . b b b b b b . .'
        '. . a a a a a a . .';
```

Car le nouveau bloc Banner me semble + étaler que sur l'integration

Anne-Gaelle-S replied:
Oui en effet, maintenant la nouvelle banner prends en largeur (surtout sur les grands écrans).
Mais si je me trompes pas c'est aussi le but de la refonte de pix-site (adapter le site aux plus grands écrans), et c'est ce que souhaitais @QuentinChapelain-ui.

**Les conséquences :**
Il ne faudrait utiliser la nouvelle banner **uniquement** lorsque la refonte totale de pix-site soit finie.
Ce qui, une fois fait, permet alors d'aligner le menu / la banner / les autres blocs de contenus.

Peut être que ces conséquences ne sont pas souhaitables (on aurait besoin de la nouvelle bannière avec les anciens format) ?

MelanieMEB reviewed (approved):
Globalement d'accord mais je me demande si la grille proposé n'est pas trop en largeur par rapport au composant Banner actuellement refait.

sbedeau commented on assets/scss/globals/_container.scss:
```diff
@@ -29,4 +29,72 @@
+  }
+}
+
+@include device-is('desktop') {
+  .row-block {
+    grid-template-areas:
+      ". a a a a . b b b .";
+
+    &--reverse {
+      grid-template-areas:
+        ". b b b . a a a a .";
+    }
+  }
+}
```
Je ne comprend pas bien la plus-value des colonnes vides par rapport à des margins. J'ai d'ailleurs le sentiment que l'on mélange deux entités, les margins ne semblent pas être des colonnes. 
Est-ce que tu peux m'en dire plus ?

```suggestion
@include device-is('desktop') {
  .row-block {
    grid-template-areas:
      "a a a a . b b b";

    &--reverse {
      grid-template-areas:
        "b b b . a a a a";
    }
  }
}
```

Anne-Gaelle-S replied:
Yess bonne question ! 
Je me suis posée la question aussi est-ce qu'il était mieux de partir sur des colonnes vides ou des margins ...
J'avais opté pour des colonnes vides (quitte à utiliser les grids les mettres sur toute la largeur de la page) mais en réalité ça n'a peut être pas beaucoup de sens en effet. 

Je vais tenter de faire un commit qui change les colonnes vides en margin et voir si c'est ok de changer :)

Anne-Gaelle-S commented on assets/scss/globals/_container.scss:
```diff
@@ -29,4 +29,72 @@
+      ". a a a a a . b b b b b b .";
+    grid-gap: 24px;
+
+    &--reverse {
+      grid-template-areas:
+        ". b b b b b . a a a a a a .";
+    }
```
Alors justement non je crois que c'est bien `". b b b b b . a a a a a a .";` qu'il faut mettre ici.
L'area `a` représentant le contenu "important" (le texte par exemple) et l'area `b` représentant le contenu accompagnant (une image par exemple), il faut que le texte prenne plus de largeur que l'image. 
Comme sur cette maquette : 
<img width="996" alt="image" src="https://user-images.githubusercontent.com/38167520/90254154-9a877d80-de42-11ea-9da1-cc053d25334d.png">
(on a 5 colonnes pour l'image , soit 5 `b` et 6 colonnes pour le texte, soit 6 `a`)

Si tu veux @sbedeau je suis dispo pour qu'on expérimente ça ensemble en live et qu'on regarde les différences :)

Anne-Gaelle-S committed:
Change grid gap into margin
